README.md

```diff
@@ -12,61 +12,94 @@
+
+### Installing WSL (Windows Users)
+If you're using Windows, you will likely want to install [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install) and use it for development. You can find in-house WSL setup instructions [here](https://github.com/badgerloop-software/github-tutorial?tab=readme-ov-file#wsl-and-ubuntu-installation-and-setup).
+
+### Installing Dependencies
+
+Before continuing, ensure that you have the following installed:
+- [node.js](https://nodejs.org/en/download) - Version 20.18.2 (LTS)
+  - [nvm](https://github.com/nvm-sh/nvm) is useful for managing node versions for multiple projects
+- [Python 3](https://www.python.org/downloads/release/python-3104/) - Version 3.10.\*
+- [Poetry](https://python-poetry.org/docs/#installing-with-the-official-installer)
+- [Docker Desktop](https://docs.docker.com/get-docker/)
+  - If using Linux, you can instead install only [Docker Engine](https://docs.docker.com/engine/install/) if you don't want the desktop application
+- [Redis Stack Server](https://redis.io/docs/latest/operate/oss_and_stack/install/install-stack/)
+  - If you want to run the server natively on Ubuntu 24.04, you may need to [install with snap](https://redis.io/docs/latest/operate/oss_and_stack/install/install-stack/linux/#on-ubuntu-with-snap)
+  - If you have installation issues or do not want to install it on your system, you can instead run a Docker container as needed with `docker run -d --name redis-stack-server -p 6379:6379 redis/redis-stack-server:latest`. See [here](https://redis.io/docs/latest/operate/oss_and_stack/install/install-stack/docker/) for more info
+
+### Cloning the Repository and Initializing the Submodules
 
 0. If you don't already have an SSH key, [generate a new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) (only the steps under "Generating a new SSH key" are required) and [add it to your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account).
-1. Once you have an SSH key, clone this repository to your computer. Make sure to clone it using SSH (when you go to copy the clone link, there will be an SSH option above the link).
-2. Next, `cd` into the `chase-car-dashboard` repository and run `git submodule update --init`.
+0. Once you have an SSH key, clone this repository to your computer. Make sure to clone it using SSH (when you go to copy the clone link, there will be an SSH option above the link).
+0. Next, `cd` into the `chase-car-dashboard` repository and run `git submodule update --init`.
 
-## Updating Data
+### Setting up Python Environment
+
+We use Poetry for managing our backend Python environment. To setup a Poetry environment for this project, `cd` into `Backend/` and run `poetry install`. When in the `Backend/` directory, run Python commands using `poetry run <command>`.
```

Comment:
Maybe mention `poetry shell` as well? I've found that more convenient sometimes

Reply:
Noted in the README